src/renderer/dx/DxFontRenderData.cpp

```diff
@@ -441,6 +453,201 @@ try
+// Return value:
+// - The float value corresponding to the passed in fontStretch
+float DxFontRenderData::FontStretchToWidthAxisValue(DWRITE_FONT_STRETCH fontStretch) noexcept
+{
+    if (gsl::narrow_cast<size_t>(fontStretch) > fontStretchEnumToVal.size())
+    {
+        fontStretch = DWRITE_FONT_STRETCH_NORMAL;
+    }
+
+    return til::at(fontStretchEnumToVal, fontStretch);
+}
+
+// Method Description:
+// - Converts a DWRITE_FONT_STYLE enum into the corresponding float value to
+//   create a DWRITE_FONT_AXIS_VALUE with
+// Arguments:
+// - fontStyle: the old DWRITE_FONT_STYLE enum to be converted into an axis value
+// Return value:
+// - The float value corresponding to the passed in fontStyle
+float DxFontRenderData::FontStyleToSlantFixedAxisValue(DWRITE_FONT_STYLE fontStyle) noexcept
+{
+    // Both DWRITE_FONT_STYLE_OBLIQUE and DWRITE_FONT_STYLE_ITALIC default to having slant.
+    // Though an italic font technically need not have slant (there exist upright ones), the
+    // vast majority of italic fonts are also slanted. Ideally the slant comes from the
+    // 'slnt' value in the STAT or fvar table, or the post table italic angle.
+
+    if (gsl::narrow_cast<size_t>(fontStyle) > fontStyleEnumToVal.size())
+    {
+        fontStyle = DWRITE_FONT_STYLE_NORMAL;
+    }
+
+    return til::at(fontStyleEnumToVal, fontStyle);
+}
+
+// Method Description:
+// - Converts a float fontSize into the corresponding float value to create a DWRITE_FONT_AXIS_VALUE with
+// Arguments:
+// - fontSize: the old float value to be converted into an axis value
+// Return value:
+// - The float value corresponding to the passed in fontSize
+float DxFontRenderData::DIPsToPoints(const float fontSize) noexcept
+{
+    return fontSize * (72.0f / 96.0f);
+}
+
+// Method Description:
+// - Fill any missing axis values that might be known but were unspecified, such as omitting
+//   the 'wght' axis tag but specifying the old DWRITE_FONT_WEIGHT enum
+// Arguments:
+// - fontWeight: the old DWRITE_FONT_WEIGHT enum to be converted into an axis value
+// - fontStretch: the old DWRITE_FONT_STRETCH enum to be converted into an axis value
+// - fontStyle: the old DWRITE_FONT_STYLE enum to be converted into an axis value
+// - fontSize: the number to convert into an axis value
+// - format: the IDWriteTextFormat3 to get the defined axes from
+// Return value:
+// - The fully formed axes vector
+std::vector<DWRITE_FONT_AXIS_VALUE> DxFontRenderData::GetAxisVector(const DWRITE_FONT_WEIGHT fontWeight,
+                                                                    const DWRITE_FONT_STRETCH fontStretch,
+                                                                    const DWRITE_FONT_STYLE fontStyle,
+                                                                    const float fontSize,
+                                                                    IDWriteTextFormat3* format)

```

Comment:
is TextFormat3 valid on Windows7 or with the "Platform Update"?

Reply:
Or: we're sure not to call this where it isn't supported?

Reply:
Yep we only call this if we have access to an IDWriteTextFormat3, I'll add it to the Method Description to be more clear about it

Reply:
(nit: not in the method description yet)

Reply:
added!